E分析、N分析、Q分析基本转换

jinmu1 opened the pull request:
1.将EIQ的E分析、N分析、Q分析的基本计算方法写在EIQClassifier
2.增加了测试类测试方法

roundpineapple committed:
去掉第二组表格的checkbox

roundpineapple committed:
Merge pull request #35 from jinmu1/fix_second_table

去掉第二组表格的checkbox

roundpineapple commented on ruoyi-admin/src/main/java/com/ruoyi/baidie/EIQClassifier.java:
```diff
@@ -0,0 +1,111 @@
+        //step1: 将日期标准化，防止同一天出现两个值
+        List<EIQBasicTable> basicTableList = new ArrayList<>();
+        for(EIQBasicTable basicTable:eiqBasicTableList){
+            basicTable.setDeliveryDate(standardizedDate(basicTable.getDeliveryDate()));
```
你写一个copy函数吧，这样还是在改原始数据

roundpineapple commented on ruoyi-admin/src/main/java/com/ruoyi/baidie/EIQClassifier.java:
```diff
@@ -0,0 +1,111 @@
+        //step3: 统计每个出库日期的数据，然后升序排序
+        final List<EIQAnalysisTable> EIQAnalysisTableSortedByDeliveryDate =
+                dataGroupByDeliveryDate.entrySet().stream()
+                        .map(entry -> calculateOutboundInfo(entry.getKey(), entry.getValue()))
```
函数名请更新

roundpineapple replied:
请把`calculateOutboundInfo`重命名为`calculateENQ`

roundpineapple commented on ruoyi-admin/src/main/java/com/ruoyi/baidie/EIQClassifier.java:
```diff
@@ -0,0 +1,111 @@
+     * @param eiqBasicTableList 导入的基本数据类
+     * @return 返回处理过的EIQ分析的值
+     */
+    private static EIQAnalysisTable calculateOutboundInfo(Date key, List<EIQBasicTable> eiqBasicTableList) {
```
函数名请更新，是复制粘贴的

roundpineapple commented on ruoyi-admin/src/main/java/com/ruoyi/baidie/EIQClassifier.java:
```diff
@@ -0,0 +1,111 @@
+     * @param deliveryDate
+     * @return
+     */
+    private static Date  standardizedDate(Date deliveryDate){
```
重命名为 normalizeDate

roundpineapple commented on ruoyi-admin/src/main/java/com/ruoyi/baidie/EIQClassifier.java:
```diff
@@ -0,0 +1,111 @@
+     * @return
+     */
+    private static Date  standardizedDate(Date deliveryDate){
+        Instant instant = deliveryDate.toInstant();
```
整个函数改为

`DateUtils.truncate(deliveryDate, Calendar.DATE);`

roundpineapple commented on ruoyi-admin/src/main/java/com/ruoyi/baidie/EIQClassifier.java:
```diff
@@ -0,0 +1,111 @@
+     * @param day 日
+     * @return 某一天的date格式
+     */
+    private static Date createDate(int year, int month, int day) {
```
不需要这个函数了

roundpineapple commented on ruoyi-admin/src/test/java/com/ruoyi/baidie/EIQClassifierTest.java:
```diff
@@ -0,0 +1,125 @@
+
+public class EIQClassifierTest {
+    private List<EIQBasicTable> testEIQBasicTable;
+    private List<Date> testSortByDates;
```
18行到21行都放到`getEIQAnalysisTableTest`里面吧，不然这个类好长。

17行可以保留

roundpineapple replied:
这个没有resolve

jinmu1 committed:
1.将EIQ的E分析、N分析、Q分析的基本计算方法写在EIQClassifier
2.增加了测试类测试方法

roundpineapple commented on ruoyi-admin/src/main/java/com/ruoyi/baidie/EIQClassifier.java:
```diff
@@ -0,0 +1,75 @@
+        List<EIQBasicTable> standardizedList = eiqBasicTableList.stream()
+                .map(basicTable -> {
+                    EIQBasicTable newBasicTable = new EIQBasicTable();
+                    newBasicTable.setDeliveryDate(normalizeDate(basicTable.getDeliveryDate()));
```
忘记了复制原来的数据去新的entry, 比如deliveryQuantity。
你现在的测试应该是不通过的。

jinmu1 committed:
1.将参数换行，规范化命名

roundpineapple commented on ruoyi-admin/src/main/java/com/ruoyi/baidie/EIQClassifier.java:
```diff
@@ -0,0 +1,87 @@
+     */
+    public static List<EIQAnalysisTable> getEIQAnalysisTable(List<EIQBasicTable> eiqBasicTableList){
+        //step1: 将日期标准化，防止同一天出现两个值
+        List<EIQBasicTable> standardizedList = eiqBasicTableList.stream()
```
final

roundpineapple commented on ruoyi-admin/src/main/java/com/ruoyi/baidie/EIQClassifier.java:
```diff
@@ -0,0 +1,87 @@
+    private static EIQAnalysisTable calculateOutboundInfo(Date key,
+                                                          List<EIQBasicTable> eiqBasicTablesGroupByDate) {
+        EIQAnalysisTable eiqAnalysisTable = new EIQAnalysisTable();
+        Map<String, List<EIQBasicTable>> collect = eiqBasicTablesGroupByDate.stream()
```
去掉这一行，直接

`final int numOrders = eiqBasicTablesGroupByDate.stream().collect(Collectors.groupingBy(EIQBasicTable::getOrderNumber)).size()`

roundpineapple commented on ruoyi-admin/src/main/java/com/ruoyi/baidie/EIQClassifier.java:
```diff
@@ -0,0 +1,87 @@
+        Map<String, List<EIQBasicTable>> collect = eiqBasicTablesGroupByDate.stream()
+                .collect(Collectors.groupingBy(EIQBasicTable::getOrderNumber));
+        eiqAnalysisTable.setDate(key);
+        eiqAnalysisTable.setEAnalysis(collect.size());
```
`eiqAnalysisTable.setEAnalysis(numOrders);`

roundpineapple reviewed (approved):
请按照建议更改然后merge

jinmu1 committed:
EI分析代码
1.修改了了命名方式
2.代码格式
3.删除多余的空格
1.修改了数据格式 将int 改为 Integer
2.修改命名方法
1.修改数据类型
2.修改命名方式
3.规范化代码格式
